src/StreamFactory/DiactorosStreamFactory.php

```diff
@@ -24,7 +24,7 @@ public function createStream($body = null)
             } else {
                 $stream = new Stream('php://memory', 'rw');
 
-                if (null !== $body) {
+                if (null !== $body || '' !== $body) {
```

Comment:
Also this doesn't work if body is an object and casting the object to string return an empty value. We should do the following : 

```
$content = (string) $body;

if (strlen($content) > 0) {
    $stream->write($content);
}

return $stream
```